### ISSUE TITLE
Multiple upload listeners and TomedLocationTasks starting

### ISSUE THREAD

**Issue by llayman** [FraunhoferCESE/madcap#95]
Some combination of putting the app in the background, closing the app, and/or clicking the notification appears to create duplicate listeners in the DatacollectionService. 

Try:
* starting MADCAP
* closing MADCAP (MADCAP will automatically restart)
* clicking the MADCAP notification, which starts the mainActivity.

I think you should see there are 4 listeners in the data collection service:

```
2-01 14:25:58.655 12590-12590/org.fraunhofer.cese.madcap D/WelcomeActivity: onStart.onConnected: successfully connected to Google Play Services.
12-01 14:25:58.671 12590-12590/org.fraunhofer.cese.madcap D/GoogleSignInApiImpl: getSavedSignInResultIfEligible
12-01 14:25:58.681 12590-12590/org.fraunhofer.cese.madcap D/GoogleSignInApiImpl: trySilentSignIn
12-01 14:25:58.688 12590-12590/org.fraunhofer.cese.madcap D/MADCAP Auth Manager: silentLogin: Immediate results are not evailable.
12-01 14:25:58.900 12590-12590/org.fraunhofer.cese.madcap D/MADCAP Auth Manager: silentLogin: Received asynchronous login result. Code: 0, message: null
12-01 14:25:58.900 12590-12590/org.fraunhofer.cese.madcap D/WelcomeActivity: User successfully signed in and authenticated to MADCAP.
12-01 14:25:58.917 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: OnStartCommand
12-01 14:25:58.930 12590-12590/org.fraunhofer.cese.madcap D/LocationServiceStatusReceiver: Check for Location preference status
12-01 14:25:58.930 12590-12590/org.fraunhofer.cese.madcap D/LocationServiceStatusReceiver: Location status updated:ON
12-01 14:25:58.932 12590-12590/org.fraunhofer.cese.madcap I/CacheFactory: CACHED Type: LocationService Data: {"LocationServiceStatus": ON} User: 111691272636258957769 Time: 1480620358930 ID:e1966943-d1a5-4797-96bf-29768857746d
12-01 14:25:58.932 12590-12590/org.fraunhofer.cese.madcap W/GmsClientEvents: registerConnectionCallbacks(): listener org.fraunhofer.cese.madcap.issuehandling.GoogleApiClientConnectionIssueManager@c2cdaf8 is already registered
12-01 14:25:58.932 12590-12590/org.fraunhofer.cese.madcap W/GmsClientEvents: registerConnectionFailedListener(): listener org.fraunhofer.cese.madcap.issuehandling.GoogleApiClientConnectionIssueManager@c2cdaf8 is already registered
12-01 14:25:58.934 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: numListeners: 4
12-01 14:25:58.936 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: LocationListener started listening
12-01 14:25:58.936 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: numListeners: 4
12-01 14:25:58.936 12590-12590/org.fraunhofer.cese.madcap D/TimedApplicationTask: API level 24
12-01 14:25:58.937 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: ApplicationsListener started listening
12-01 14:25:58.937 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: numListeners: 4
12-01 14:25:58.937 12590-13088/org.fraunhofer.cese.madcap D/TimedApplicationTask: started doInBackground
12-01 14:25:58.937 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: LocationListener started listening
12-01 14:25:58.937 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: numListeners: 4
12-01 14:25:58.937 12590-12590/org.fraunhofer.cese.madcap D/TimedApplicationTask: API level 24
12-01 14:25:58.937 12590-12590/org.fraunhofer.cese.madcap D/Madcap DataColl Service: ApplicationsListener started listening
12-01 14:25:58.941 12590-13087/org.fraunhofer.cese.madcap D/TimedLocationTask: Permission granted
```

**Comment by mmueller301** [comment]
Works fine before your Fixes #94 and Fixes #52 commits (d29b0935fd97f30b7822875a5685a533a6f57be1 and 604392406343bd0094e1d19816de45308baa9609) @llayman . So I guess it must have been introduced there.

**Comment by mmueller301** [comment]
Commit d29b0935fd97f30b7822875a5685a533a6f57be1 is the one. We have to reevaluate your improvements to not get lost in logic @llayman
